components/brave_wallet/browser/eth_json_rpc_controller.cc

```diff
@@ -62,67 +62,64 @@ bool GetUseStagingInfuraEndpoint() {
+    mojom::Network network,
     scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory)
     : api_request_helper_(GetNetworkTrafficAnnotationTag(), url_loader_factory),
       network_(network),
-      observers_(new base::ObserverListThreadSafe<
-                 BraveWalletProviderEventsObserver>()),
       weak_ptr_factory_(this) {
   SetNetwork(network);
 }
 
 EthJsonRpcController::~EthJsonRpcController() {}
 
-void EthJsonRpcController::AddObserver(
-    BraveWalletProviderEventsObserver* observer) {
-  observers_->AddObserver(observer);
+mojo::PendingRemote<mojom::EthJsonRpcController>
+EthJsonRpcController::MakeRemote() {
+  mojo::PendingRemote<mojom::EthJsonRpcController> remote;
+  receivers_.Add(this, remote.InitWithNewPipeAndPassReceiver());
+  return remote;
 }
 
-void EthJsonRpcController::RemoveObserver(
-    BraveWalletProviderEventsObserver* observer) {
-  observers_->RemoveObserver(observer);
+void EthJsonRpcController::AddObserver(
+    ::mojo::PendingRemote<mojom::EthJsonRpcControllerObserver> observer) {
+  observers_.Add(std::move(observer));
 }
 
 void EthJsonRpcController::Request(const std::string& json_payload,
-                                   URLRequestCallback callback,
-                                   bool auto_retry_on_network_change) {
+                                   bool auto_retry_on_network_change,
+                                   RequestCallback callback) {
   api_request_helper_.Request("POST", network_url_, json_payload,
                               "application/json", auto_retry_on_network_change,
                               std::move(callback));
 }
 
-Network EthJsonRpcController::GetNetwork() const {
-  return network_;
-}
-
-GURL EthJsonRpcController::GetNetworkURL() const {
-  return network_url_;
+void EthJsonRpcController::GetNetwork(
+    mojom::EthJsonRpcController::GetNetworkCallback callback) {
+  std::move(callback).Run(network_);
 }
 
-void EthJsonRpcController::SetNetwork(Network network) {
+void EthJsonRpcController::SetNetwork(mojom::Network network) {
   std::string subdomain;
   network_ = network;
   switch (network) {
-    case Network::kMainnet:
+    case brave_wallet::mojom::Network::Mainnet:
       subdomain = "mainnet";
       break;
-    case Network::kRinkeby:
+    case brave_wallet::mojom::Network::Rinkeby:
       subdomain = "rinkeby";
       break;
-    case Network::kRopsten:
+    case brave_wallet::mojom::Network::Ropsten:
       subdomain = "ropsten";
       break;
-    case Network::kGoerli:
+    case brave_wallet::mojom::Network::Goerli:
       subdomain = "goerli";
       break;
-    case Network::kKovan:
+    case brave_wallet::mojom::Network::Kovan:
       subdomain = "kovan";
       break;
-    case Network::kLocalhost:
+    case brave_wallet::mojom::Network::Localhost:
       network_url_ = GURL("http://localhost:8545");
       return;
-    case Network::kCustom:
+    case brave_wallet::mojom::Network::Custom:
       return;
```

Comment:
calling this with `Custom` will set the network, but will leave the `network_url_` at whatever value it was before this

Reply:
did you want me to DCHECK to make sure Custom is not used in this API?